app/web_framework/cluster_master_application.py

```diff
@@ -41,7 +42,8 @@ def __init__(self, cluster_master):
                     ]),
                     RouteNode(r'queue', _QueueHandler),
                     RouteNode(r'slave', _SlavesHandler, 'slaves').add_children([
-                        RouteNode(r'(\d+)', _SlaveHandler, 'slave')
+                        RouteNode(r'(\d+)', _SlaveHandler, 'slave'),
```

Comment:
Good thinking.  I'm adding this route and moving some of the common shutdown logic to ClusterMaster